packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -200,15 +202,15 @@ export class Table extends Construct {
   private readonly secondaryIndexNames: string[] = [];
   private readonly nonKeyAttributes: string[] = [];
 
-  private tablePartitionKey?: Attribute;
-  private tableSortKey?: Attribute;
+  private readonly tablePartitionKey: Attribute;
+  private readonly tableSortKey: Attribute | undefined;
```

Comment:
Any particular reason not to `private readonly tableSortKey?: Attribute`? (I am not asking for this to be changed - just curious).

Reply:
I was sure you'll be proud of me. I think it's more accurate: sort key is not really optional, it's just that it can be empty. But you must specify it.

Reply:
Subtle semantic distinction, I guess. Funny thing is - not all code paths in the constructor assign a value to `this.tableSortKey`, so it effectively *is* optional :)